polaris-core/src/main/java/org/apache/polaris/core/persistence/cache/EntityCache.java

```diff
@@ -365,50 +230,46 @@ && isNewer(existingCacheEntry, existingCacheEntryByName)) {
       long entityCatalogId,
       long entityId,
       PolarisEntityType entityType) {
-
-    // if it exists, we are set
-    ResolvedPolarisEntity entry = this.getEntityById(entityId);
-    final boolean cacheHit;
-
-    // we need to load it if it does not exist
-    if (entry == null) {
-      // this is a miss
-      cacheHit = false;
-
-      // load it
-      ResolvedEntityResult result =
-          polarisMetaStoreManager.loadResolvedEntityById(
-              callContext, entityCatalogId, entityId, entityType);
-
-      // not found, exit
-      if (!result.isSuccess()) {
-        return null;
-      }
-
-      // if found, setup entry
-      callContext.getDiagServices().checkNotNull(result.getEntity(), "entity_should_loaded");
-      callContext
-          .getDiagServices()
-          .checkNotNull(result.getEntityGrantRecords(), "entity_grant_records_should_loaded");
-      entry =
-          new ResolvedPolarisEntity(
-              callContext.getDiagServices(),
-              result.getEntity(),
-              result.getEntityGrantRecords(),
-              result.getGrantRecordsVersion());
-
-      // the above loading could take a long time so check again if the entry exists and only
-      this.cacheNewEntry(entry);
+    final AtomicBoolean cacheHit = new AtomicBoolean(true);
```

Comment:
Nit: `final` is syntactic sugar.
Could just be 
```suggestion
    var cacheHit = new AtomicBoolean(true);
```

Reply:
Nope. That's why it's a "nit".